lib/bolt/inventory.rb

```diff
@@ -126,12 +136,10 @@ def update_target(target)
         data['config'] = {}
       end
 
-      unless data['vars']
-        @logger.debug("Did not find any variables for #{target.name} in inventory")
-        data['vars'] = {}
-      end
-
-      set_vars_from_hash(target.name, data['vars'])
+      # These should only get set from the inventory if they have not yet
+      # been instantiated
+      set_vars_from_hash(target.name, data['vars']) unless @target_vars[target.name]
```

Comment:
So this initializes them the 1st time from the inventory data, then all subsequent lookups will use whatever latest is. 👍

Reply:
Yep :P

Reply:
Is there a test that covers this particular fix?

Reply:
Ah, I see it in spec/pal/vars_spec.rb.